src/kernel/arch/x86/pit.zig

```diff
@@ -0,0 +1,288 @@
+        // The fastest possible frequency if frequency is too high
+        reload_value = 1;
+
+        if (freq < MAX_FREQUENCY) {
+            reload_value = MAX_FREQUENCY / freq;
+            var rem: u32 = MAX_FREQUENCY % freq;
+
+            // Is the remainder more than half
+            if (rem > MAX_FREQUENCY / 2) {
+                // Round up
+                reload_value += 1;
+            }
+        }
+    }
+
+    // Get the u16 version as this is what will be loaded into the PIT
+    var reload_val_16: u16 = @truncate(u16, reload_value);
+
+    // Update the frequency with the actual one that the PIT will be using
+    var frequency: u32 = MAX_FREQUENCY / reload_value;
+    var rem: u32 = MAX_FREQUENCY % reload_value;
+
+    // Is the remainder more than half
+    if (rem > MAX_FREQUENCY / 2) {
+        // Round up
+        frequency += 1;
+    }
+
+    // Calculate the amount of time between IRQs
+    time_ms = reload_value * u32(1000);
+    time_under_1_ms = time_ms % MAX_FREQUENCY;
+    time_ms /= MAX_FREQUENCY;
+
+    comptime const port: u16 = switch (counter) {
+        OCW_SELECT_COUNTER_0 => COUNTER_0_REGISTER,
+        OCW_SELECT_COUNTER_1 => COUNTER_1_REGISTER,
+        OCW_SELECT_COUNTER_2 => COUNTER_2_REGISTER,
+        else => unreachable,
+    };
+
+    switch (counter) {
+        OCW_SELECT_COUNTER_0 => current_freq_0 = frequency,
+        OCW_SELECT_COUNTER_1 => current_freq_1 = frequency,
+        OCW_SELECT_COUNTER_2 => current_freq_2 = frequency,
+        else => unreachable,
+    }
+
+    var command: u8 = 0;
+    command |= mode;
+    command |= OCW_READ_LOAD_DATA;
+    command |= counter;
+
+    sendCommand(command);
+
+    sendDateToCounter(port, @truncate(u8, reload_val_16));
+    sendDateToCounter(port, @truncate(u8, reload_val_16 >> 8));
+
+    // Reset the counter ticks
+    switch (counter) {
+        OCW_SELECT_COUNTER_0 => ticks = 0,
+        OCW_SELECT_COUNTER_1 => ram_ticks = 0,
+        OCW_SELECT_COUNTER_2 => speaker_ticks = 0,
+        else => unreachable,
+    }
+}
+
+///
+/// A simple wait that used the PIT to wait a number of ticks.
+///
+/// Arguments:
+///     IN milliseconds: u32 - The number of ticks to wait.
+///
+pub fn waitTicks(ticks_to_wait: u32) void {
+    const wait_ticks = ticks + ticks_to_wait;
+    while (ticks < wait_ticks) {
+        arch.enableInterrupts();
+        arch.halt();
+        arch.disableInterrupts();
+    }
+    arch.enableInterrupts();
+}
+
+///
+/// Get the number of ticks that have passed when the PIT was initiated.
+///
+/// Return:
+///     Number of ticks passed.
+///
+pub fn getTicks() u32 {
+    return ticks;
+}
+
+///
+/// Get the frequency the PIT is ticking at.
+///
+/// Return:
+///     The frequency the PIT is running at
+///
+pub fn getFrequency() u32 {
+    return current_freq_0;
+}
+
+///
+/// Initialise the PIT with a handler to IRQ 0.
+///
+pub fn init() void {
+    // Set up counter 0 at 1000hz in a square wave mode counting in binary
```

Comment:
If you rebase on top of the logging PR you could log something like "Init pit" (xD) and change the tty.print below to a log statement.

Reply:
Sure thing